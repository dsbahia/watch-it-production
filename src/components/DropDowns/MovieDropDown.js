import React from "react";
import Dropdown from "react-bootstrap/Dropdown";
import DropdownButton from "react-bootstrap/DropdownButton";

function MovieDropDown() {
  return (
    <DropdownButton id="item-movie-dropdown" title="Movies">
      <Dropdown.Item data-testid="top-rated-link" href="/top-rated-movies">
        Top Rated
      </Dropdown.Item>
      <Dropdown.Item data-testid="upcoming-link" href="/upcoming-movies">
        Upcoming
      </Dropdown.Item>
    </DropdownButton>
  );
}

export default MovieDropDown;
