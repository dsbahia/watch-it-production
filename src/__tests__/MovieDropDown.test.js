import React from "react";
import { fireEvent, render } from "@testing-library/react";
import MovieDropDown from "../components/DropDowns/MovieDropDown";

describe("MovieDropdown", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<MovieDropDown />);
    expect(asFragment).toMatchSnapshot();
  });

  it("displays 'Movies' title", () => {
    const { getByText } = render(<MovieDropDown />);
    expect(getByText("Movies")).toBeInTheDocument();
  });

  it("displays dropdowns with correct paths", () => {
    const { getByTestId, getByText } = render(<MovieDropDown />);
    const dropdownButton = getByText("Movies");

    fireEvent.click(dropdownButton);

    const topRated = getByTestId("top-rated-link");
    const upcoming = getByTestId("upcoming-link");

    expect(topRated).toHaveAttribute("href", "/top-rated-movies");
    expect(upcoming).toHaveAttribute("href", "/upcoming-movies");
  });
});
