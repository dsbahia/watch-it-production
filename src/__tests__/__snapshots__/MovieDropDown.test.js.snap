// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDropdown renders correctly 1`] = `[Function]`;
